### ISSUE TITLE
Consider contributing some rules up to typescript-eslint?

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [danielnixon/eslint-plugin-total-functions#665]
Hi! I found this plugin recently and really like it 😄 thanks for making it! Some of the rules seem pretty generally useful:

* `total-functions/no-unsafe-type-assertion`: the stronger form of `objectLiteralTypeAssertions` might be good to pull in as an option for `@typescript-eslint/consistent-type-assertion`?
* `total-functions/no-unsafe-readonly-mutable-assignment` & `total-functions/no-unsafe-mutable-readonly-assignment`: I've been hit by this before!
* `total-functions/no-unsafe-optional-property-assignment`: Also broadly useful, once #83 lands

Would you be open to having these rules taken into typescript-eslint itself? We'd want to have issues filed on github.com/typescript-eslint/typescript-eslint for each to discuss, but since they've been proven useful by this plugin I'd be surprised if there were any holdups. 😄

**Comment by JoshuaKGoldberg** [comment]
Filed issues over on github.com/typescript-eslint/typescript-eslint! 🚀 

Note that I didn't file one for `total-functions/no-unsafe-mutable-readonly-assignment`. Per the README.md it's quite noisy in practice. If someone else feels strongly that it's useful and should be added to typescript-eslint core, I'd encourage them to file an issue too 😄.

**Comment by JoshuaKGoldberg** [comment]
Closing out my old issues for repos I no longer have context on. If anybody has a need for what this issue was asking about, I'd encourage them/you to file a new issue. Cheers!
